From 0bcc41f3617e9041513a779a8a54f7fecae1efcd Mon Sep 17 00:00:00 2001
From: Tianling Shen <cnsztl@immortalwrt.org>
Date: Wed, 11 Mar 2026 15:37:12 +0800
Subject: [PATCH] rust: Update to 1.94.0

Changelog:
- https://github.com/rust-lang/rust/releases/tag/1.91.1
- https://github.com/rust-lang/rust/releases/tag/1.92.0
- https://github.com/rust-lang/rust/releases/tag/1.93.0
- https://github.com/rust-lang/rust/releases/tag/1.93.1
- https://github.com/rust-lang/rust/releases/tag/1.94.0

Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>
(cherry picked from commit ff826adfa8a3d8545f8589b0949cae0de55efc02)
---
 lang/rust/Makefile                                        | 6 +++---
 lang/rust/patches/0001-Update-xz2-and-use-it-static.patch | 4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/lang/rust/Makefile b/lang/rust/Makefile
index 710d49925ab9dc..7ab3d1333d9dc2 100644
--- a/lang/rust/Makefile
+++ b/lang/rust/Makefile
@@ -5,12 +5,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=rust
-PKG_VERSION:=1.90.0
-PKG_RELEASE:=2
+PKG_VERSION:=1.94.0
+PKG_RELEASE:=1
 
 PKG_SOURCE:=rustc-$(PKG_VERSION)-src.tar.xz
 PKG_SOURCE_URL:=https://static.rust-lang.org/dist/
-PKG_HASH:=6bfeaddd90ffda2f063492b092bfed925c4b8c701579baf4b1316e021470daac
+PKG_HASH:=0b53ae34f5c0c3612cfe1de139f9167a018cd5737bc2205664fd69ba9b25f600
 HOST_BUILD_DIR:=$(BUILD_DIR)/host/rustc-$(PKG_VERSION)-src
 
 PKG_MAINTAINER:=Luca Barbato <lu_zero@luminem.org>
diff --git a/lang/rust/patches/0001-Update-xz2-and-use-it-static.patch b/lang/rust/patches/0001-Update-xz2-and-use-it-static.patch
index d3625c9bc4c73a..c90c76cfa87b37 100644
--- a/lang/rust/patches/0001-Update-xz2-and-use-it-static.patch
+++ b/lang/rust/patches/0001-Update-xz2-and-use-it-static.patch
@@ -9,7 +9,7 @@ Subject: [PATCH] Update xz2 and use it static
 
 --- a/src/bootstrap/Cargo.toml
 +++ b/src/bootstrap/Cargo.toml
-@@ -55,7 +55,7 @@ tar = "0.4"
+@@ -54,7 +54,7 @@ tar = { version = "0.4.44", default-feat
  termcolor = "1.4"
  toml = "0.5"
  walkdir = "2.4"
@@ -17,4 +17,4 @@ Subject: [PATCH] Update xz2 and use it static
 +xz2 = { version = "0.1", features = ["static"] }
  
  # Dependencies needed by the build-metrics feature
- sysinfo = { version = "0.36.0", default-features = false, optional = true, features = ["system"] }
+ sysinfo = { version = "0.37.0", default-features = false, optional = true, features = ["system"] }

